pocs/linux/kernelctf/CVE-2024-41009_lts_cos/exploit/cos-105-17412.370.23/exploit.c

```diff
@@ -0,0 +1,589 @@
+
+#define BPF_LSH_IMM(DST, IMM) \
+	BPF_RAW_INSN(BPF_ALU64 | BPF_LSH | BPF_K, DST, 0, 0, IMM)
+
+#define BPF_ALU64_IMM(OP, DST, IMM) \
+	BPF_RAW_INSN(BPF_ALU64 | BPF_OP(OP) | BPF_K, DST, 0, 0, IMM)
+
+#define BPF_ALU64_REG(OP, DST, SRC) \
+	BPF_RAW_INSN(BPF_ALU64 | BPF_OP(OP) | BPF_X, DST, SRC, 0, 0)
+
+#define BPF_ALU_IMM(OP, DST, IMM) \
+	BPF_RAW_INSN(BPF_ALU | BPF_OP(OP) | BPF_K, DST, 0, 0, IMM)
+
+#define BPF_JMP_IMM(OP, DST, IMM, OFF) \
+	BPF_RAW_INSN(BPF_JMP | BPF_OP(OP) | BPF_K, DST, 0, OFF, IMM)
+
+#define BPF_JMP_REG(OP, DST, SRC, OFF) \
+	BPF_RAW_INSN(BPF_JMP | BPF_OP(OP) | BPF_X, DST, SRC, OFF, 0)
+
+#define BPF_JMP32_REG(OP, DST, SRC, OFF) \
+	BPF_RAW_INSN(BPF_JMP32 | BPF_OP(OP) | BPF_X, DST, SRC, OFF, 0)
+
+#define BPF_JMP32_IMM(OP, DST, IMM, OFF) \
+	BPF_RAW_INSN(BPF_JMP32 | BPF_OP(OP) | BPF_K, DST, 0, OFF, IMM)
+
+#define BPF_EXIT_INSN() BPF_RAW_INSN(BPF_JMP | BPF_EXIT, 0, 0, 0, 0)
+
+#define BPF_LD_MAP_FD(DST, MAP_FD) \
+	BPF_LD_IMM64_RAW(DST, BPF_PSEUDO_MAP_FD, MAP_FD)
+
+#define BPF_LD_IMM64(DST, IMM) BPF_LD_IMM64_RAW(DST, 0, IMM)
+
+#define BPF_ST_MEM(SIZE, DST, OFF, IMM) \
+	BPF_RAW_INSN(BPF_ST | BPF_SIZE(SIZE) | BPF_MEM, DST, 0, OFF, IMM)
+
+#define BPF_LDX_MEM(SIZE, DST, SRC, OFF) \
+	BPF_RAW_INSN(BPF_LDX | BPF_SIZE(SIZE) | BPF_MEM, DST, SRC, OFF, 0)
+
+#define BPF_STX_MEM(SIZE, DST, SRC, OFF) \
+	BPF_RAW_INSN(BPF_STX | BPF_SIZE(SIZE) | BPF_MEM, DST, SRC, OFF, 0)
+
+#define BPF_LD_ABS(SIZE, IMM)                                     \
+	((struct bpf_insn){.code = BPF_LD | BPF_SIZE(SIZE) | BPF_ABS, \
+					   .dst_reg = 0,                              \
+					   .src_reg = 0,                              \
+					   .off = 0,                                  \
+					   .imm = IMM})
+
+#define BPF_MAP_GET(idx, dst)                                   \
+	BPF_MOV64_REG(BPF_REG_1, BPF_REG_9),                        \
+		BPF_MOV64_REG(BPF_REG_2, BPF_REG_10),                   \
+		BPF_ALU64_IMM(BPF_ADD, BPF_REG_2, -4),                  \
+		BPF_ST_MEM(BPF_W, BPF_REG_10, -4, idx),                 \
+		BPF_RAW_INSN(BPF_JMP | BPF_CALL, 0, 0, 0,               \
+					 BPF_FUNC_map_lookup_elem),                 \
+		BPF_JMP_IMM(BPF_JNE, BPF_REG_0, 0, 1), BPF_EXIT_INSN(), \
+		BPF_LDX_MEM(BPF_DW, dst, BPF_REG_0, 0),                 \
+		BPF_MOV64_IMM(BPF_REG_0, 0)
+
+#define BPF_MAP_GET_ADDR(idx, dst)                              \
+	BPF_MOV64_REG(BPF_REG_1, BPF_REG_9),                        \
+		BPF_MOV64_REG(BPF_REG_2, BPF_REG_10),                   \
+		BPF_ALU64_IMM(BPF_ADD, BPF_REG_2, -4),                  \
+		BPF_ST_MEM(BPF_W, BPF_REG_10, -4, idx),                 \
+		BPF_RAW_INSN(BPF_JMP | BPF_CALL, 0, 0, 0,               \
+					 BPF_FUNC_map_lookup_elem),                 \
+		BPF_JMP_IMM(BPF_JNE, BPF_REG_0, 0, 1), BPF_EXIT_INSN(), \
+		BPF_MOV64_REG((dst), BPF_REG_0), BPF_MOV64_IMM(BPF_REG_0, 0)
+
+#define LOG_BUF_SIZE 65536
+
+#define BPF_HEAVY_JOB                             \
```

Comment:
This isn't used right? Could you remove the parts that are unused so that it's clearer for other readers? :)